helix-core/src/main/java/org/apache/helix/task/AbstractTaskDispatcher.java

```diff
@@ -811,13 +816,23 @@ private static void addCompletedTasks(Set<Integer> set, JobContext ctx, Iterable
           || state == TaskPartitionState.DROPPED) {
         filteredTasks.add(partitionNumber);
       }
-      // Allow tasks whose assigned instances are no longer live for rescheduling
+      // Allow tasks whose assigned instances are no longer live or enable for rescheduling
       if (isTaskNotInTerminalState(state)) {
         String assignedParticipant = jobContext.getAssignedParticipant(partitionNumber);
+        final String pName = pName(jobResource, partitionNumber);
```

Comment:
NIT : Let's not have the variable name same as the function name.

Reply:
Thanks for the review. Fixed.